src/linker/Linker/TypeNameResolver.cs

```diff
@@ -13,51 +13,54 @@ public TypeNameResolver (LinkContext context)
 			_context = context;
 		}
 
-		public TypeReference ResolveTypeName (string typeNameString)
+		static TypeName TryParseTypeName (string typeNameString)
```

Comment:
Should we change this to:

```suggestion
		static bool TryParseTypeName (string typeNameString, out TypeName typeName)
```